### PR TITLE
TileManager cleanup + use make_unique

### PR DESCRIPTION
**Pull request by hjanetzek**
GetInstance did nothing but create new instances

### CODE REVIEW

File: core/src/tile/tileManager.h [hunk 1]
```diff
@@ -32,11 +32,7 @@ class TileManager {
 
 public:
 
-    /* Returns the single instance of the TileManager */
-    static std::unique_ptr<TileManager> GetInstance() {
```
Comment (tallytalwar) [inline core/src/tile/tileManager.h]: Ohh I think we needed a private static Tile manager instance for it to be truely singleton.

Comment (hjanetzek) [inline core/src/tile/tileManager.h]: And would it still be useful? This implementation does create an instance and then directly move it out as return value, so each call to GetInstance returns a new instance.

Comment (blair1618) [inline core/src/tile/tileManager.h]: You know, in retrospect I don't think we need to jump through hoops to get the compiler to enforce that `TileManager` is a singleton :)

**Comment by blair1618** [comment]
Looks good to me, cheers to c++14!

**Comment by tallytalwar** [comment]
I agree.

**Pull request merged**